api/server/router/swarm/backend.go

```diff
@@ -14,8 +16,8 @@ type Backend interface {
 	Update(uint64, types.Spec) error
 	GetServices(basictypes.ServiceListOptions) ([]types.Service, error)
 	GetService(string) (types.Service, error)
-	CreateService(types.ServiceSpec) (string, error)
-	UpdateService(string, uint64, types.ServiceSpec) error
+	CreateService(types.ServiceSpec, http.Header) (string, error)
```

Comment:
Don't use headers here. Headers should be handled in the server layer. Just pass the auth data here. Maybe makes sense to use same `types.AuthConfig` like the pull methods do.